src/soho/dropdown/soho-dropdown.component.ts

```diff
@@ -1,33 +1,49 @@
 import {
   AfterViewInit,
-  ChangeDetectionStrategy,
   Component,
   ElementRef,
   EventEmitter,
   HostBinding,
   Input,
   OnDestroy,
   Output,
+  NgZone,
+  Self,
+  Host,
+  Optional,
+  InjectionToken,
+  forwardRef,
+  AfterViewChecked
 } from '@angular/core';
-import { NgModel } from '@angular/forms';
+
+import {
+  NgModel,
+  NgControl,
+  ControlValueAccessor
+} from '@angular/forms';
 
 @Component({
   selector: 'select[soho-dropdown]', // tslint:disable-line
-  template: '<ng-content></ng-content>',
-  providers: [NgModel],
-  changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
Curious to why this is being removed; I think it is something we want to keep.

Additionally, we still need the change for @Input set source() where the source function is wrapped to run inside of Angular, is that correct?

Reply:
The JIRA was to remove the `NgModel` from the `providers` list, as this was causing problems for another team and their build.  The idea was to replace this with a custom `ControlValueAccessor` which would provide the glue between the Soho (IDS) control and Angular.  This proved complicated, as the value accessors needed to be specific for each selection type so I decided to wrap the existing ones allowing me to grab the `changeFn` to invoke from within the `changed` event.

I will add back `changeDetection: ChangeDetectionStrategy.OnPush`, I must have removed it when removing `ChangeDetectorRef` from the constructor (which I no longer used).

The problem with forcing the `source` method to run inside Angular, is that this causes all the code invoked as part of the source/response interaction also to run inside the Angular zone.  I found this caused spurious changes in any associated Form Model, and broke multi-selection.

I found that by adding a handler for the `requestEnd` I was able to force an update, such that the dropdown was updated correctly, this seemed to negate the need for the `source` delegate.

This also removed the need for the caller to invoke `markForCheck`.

I would really appreciate you giving this a go in your code, as different usages introduce different issues.